lib/MusicBrainz/Server/Data/Release.pm

```diff
@@ -245,6 +245,22 @@ sub find_by_artist
     $self->query_to_list_limited($query, $params, $limit, $offset, undef, cache_hits => 1);
 }
 
+sub find_by_artist_credit
+{
+    my ($self, $artist_credit_id, $limit, $offset) = @_;
+
+    my $query = "SELECT DISTINCT " . $self->_columns . ",
+                   date_year, date_month, date_day, area.name AS country_name,
+                   release.name COLLATE musicbrainz AS name_collate
+                 FROM " . $self->_table . "
+                 LEFT JOIN release_event ON release_event.release = release.id
+                 LEFT JOIN area ON area.id = release_event.country
+                 WHERE artist_credit = ?
+                 ORDER BY date_year, date_month, date_day,
+                   country_name, barcode, release.name COLLATE musicbrainz";
```

Comment:
Since we don't actually display the year, country, etc., I would probably remove all those joins and sorts (just sort by name like the rest). Then you can remove the `DISTINCT` too.